File: test/Automation_python_CodeGen_Quality_RealRepo_python/pipelines/Python_E2E_Existing_Pipeline_All_2/code/python_e2e_existing_pipeline_all_2/graph/WindowFunction_1.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.utils import *
from prophecy.libs import typed_lit
from python_e2e_existing_pipeline_all_2.config.ConfigStore import *
from python_e2e_existing_pipeline_all_2.udfs.UDFs import *

def WindowFunction_1(spark: SparkSession, in0: DataFrame) -> DataFrame:
    return in0\
        .withColumn(
          "last_name",
          row_number()\
            .over(Window.partitionBy(col("customer_id"), col("country_code")).orderBy(col("first_name").asc()))
        )\
        .withColumn("phone", row_number()\
        .over(Window.partitionBy(col("customer_id"), col("country_code")).orderBy(col("first_name").asc())))

File: test/Automation_scala_CodeGen_Quality_RealRepo_scala/pipelines/Scala_E2E_Existing_Pipeline_All_2/code/src/main/scala/io/prophecy/pipelines/scala_e2e_existing_pipeline_all_2/graph/WindowFunction_1.scala
package io.prophecy.pipelines.scala_e2e_existing_pipeline_all_2.graph

import io.prophecy.libs._
import io.prophecy.pipelines.scala_e2e_existing_pipeline_all_2.udfs.PipelineInitCode._
import io.prophecy.pipelines.scala_e2e_existing_pipeline_all_2.udfs.UDFs._
import io.prophecy.pipelines.scala_e2e_existing_pipeline_all_2.config.Context
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object WindowFunction_1 {

  def apply(context: Context, in: DataFrame): DataFrame = {
    import org.apache.spark.sql.expressions.{Window, WindowSpec}
    in.withColumn(
        "country_code",
        row_number().over(
          Window
            .partitionBy(col("customer_id"),
                         col("first_name"),
                         col("last_name")
            )
            .orderBy(col("phone").asc, col("email").asc)
        )
      )
      .withColumn(
        "account_open_date",
        row_number().over(
          Window
            .partitionBy(col("customer_id"),
                         col("first_name"),
                         col("last_name")
            )
            .orderBy(col("phone").asc, col("email").asc)
        )
      )
      .withColumn(
        "account_flags",
        row_number().over(
          Window
            .partitionBy(col("customer_id"),
                         col("first_name"),
                         col("last_name")
            )
            .orderBy(col("phone").asc, col("email").asc)
        )
      )
  }

}
